### PR TITLE
Autogenerate the README and a pages branch from machine-readable data

### PR DESCRIPTION
**Pull request by alimpfard**

### CODE REVIEW

File: data/yaksplained.json [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    "image": "https://cdn.discordapp.com/emojis/901168038248874045.png",
+    "remark": "Congratulations!",
+    "name": "Partyak"
+  },
+  {
+    "use_in_serenity": "U+10CD08",
+    "creation_history": "Created by Linus on October 9th, 2021. Reason: After discussing the :goldyak: origins (see above) with [Linus' blacked-out screenshot](https://discord.com/channels/830522505605283862/830525235803586570/896476288167071824), Sam asked [\":reyakted: emote wen?\"](https://discord.com/channels/830522505605283862/830525235803586570/896484164059156490) and Linus promptly created it. ",
+    "description": "(Note to the RFC editor: Insert an SCP joke here.)",
+    "image": "https://cdn.discordapp.com/emojis/896486220106956810.png",
+    "remark": "This yak is named \\[REDACTED\\], it was born on ■■■■■■■.",
```
Comment (kleinesfilmroellchen) [inline data/yaksplained.json:70]: I assume the backslashes still work in Markdown

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Build README.md
+        run: |
+          set -e
+          cat templates/README-prologue.md > README.md
+          jq 'sort_by(.name)|map(.name|"- [\(.)](#\(.))")|join("\n")' -r data/yaksplained.json >> README.md
+          echo >> README.md
+          cat templates/README-private-channels.md >> README.md
+          echo >> README.md
+          jq 'sort_by(.name) | map("### \(.name): \(.remark)\n![\(.name)](\(.image))\n\n\(.description)\n\n\(.creation_history)\n\n\(if .use_in_serenity != null then "Use in SerenityOS: `" + .use_in_serenity + "`\n" else "" end)")|join("\n")' -r data/yaksplained.json >> README.md
```
Comment (kleinesfilmroellchen) [inline .github/workflows/update.yml:47]: Is it intentional to have a backslash before the parameters (like .name, .image), but not after them?

File: data/yaksplained.json (hunk 1 above)
Comment (alimpfard) [reply data/yaksplained.json:70]: sure, give it a minute and I'll show the generated stuff.

File: .github/workflows/update.yml (hunk 2 above)
Comment (alimpfard) [reply .github/workflows/update.yml:47]: yes, that's jq's string interpolation syntax.

File: .github/workflows/update.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          git checkout pages
+          git add api
+          git commit -m 'Generate new pages'
+          git checkout main
+
+      - name: Build README.md
+        run: |
+          set -e
+          cat templates/README-prologue.md > README.md
+          jq 'sort_by(.name)|map(.name|"- [\(.)](#\(.))")|join("\n")' -r data/yaksplained.json >> README.md
```
Comment (kleinesfilmroellchen) [inline .github/workflows/update.yml:43]: The problem with this auto-generated TOC is that the text of the headers doesn't just contain the name. It also contains the remarks, and Markdown has some particular way of converting this complex special-character-containing text into an HTML fragment (or id) compatible string. AFAICT that just involves lower-casing, removing most special characters and replacing all spaces by dashes. But don't quote me. Bottom line is: these hyperlinks will not work, though I don't want to give up the remarks in the title line this quickly.

Comment (alimpfard) [reply .github/workflows/update.yml:43]: I'll deal with that somehow, you're right, they won't work like this.

**Pull request merged**